### PR TITLE
Ensure convolution weights are contiguous, fixes #4500

### LINKED ISSUE #4500: CUDA ConvTranspose double backwards disagrees with CPU
When I added more robust testing for double backwards, I discovered that the following tests started failing with giant numerical errors:

```
         - test/test_nn.py:4094 TestNN.test_ConvTranspose1d_dilated_cuda
         - test/test_nn.py:4094 TestNN.test_ConvTranspose2d_dilated_cuda
         - test/test_nn.py:4094 TestNN.test_ConvTranspose2d_groups_cuda
         - test/test_nn.py:4094 TestNN.test_ConvTranspose3d_cuda
         - test/test_nn.py:4094 TestNN.test_ConvTranspose3d_dilated_cuda
```

I am not sure why they are failing.

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by soumith** [comment]
@ezyang isn't it cleaner to just have the CuDNN bindings make weight contiguous always...?

File: aten/src/ATen/native/Convolution.cpp [hunk 1]
```diff
@@ -35,6 +35,20 @@ struct ConvParams {
...
+      << "  padding = " << IntList{params.padding}
+      << "  dilation = " << IntList{params.dilation}
+      << "  transposed = " << params.transposed
+      << "  output_padding = " << IntList{params.output_padding}
+      << "  groups = " << params.groups
+      << "  benchmark = " << params.benchmark
+      << "  deterministic = " << params.deterministic
+      << "  cudnn_enabled = " << params.cudnn_enabled
+      << "}";
+}
```
Comment (yf225) [inline aten/src/ATen/native/Convolution.cpp]: I think we can just `return out;` here to fix the Windows CI error

**Comment by ezyang** [comment]
> @ezyang isn't it cleaner to just have the CuDNN bindings make weight contiguous always...?

Yeah, I was vacillating between the two ways. I'll swap it.

> I think we can just return out; here to fix the Windows CI error

It's not just a CI error; it's a legit bug. Thanks for pointing it out!

**Review by apaszke — APPROVED** [review]

**Pull request merged**